rkt/metadata_service.go

```diff
@@ -541,12 +539,19 @@ func runMetadataService(args []string) (exit int) {
 
 	log.Print("Metadata service running...")
 
-	if err = srv.Serve(l); err != nil {
-		log.Printf("Error serving HTTP: %v", err)
+	go func() {
+		if err = srv.Serve(unixl); err != nil {
```

Comment:
is there a guarantee that this will fail immediately (if at all)? if not, surely you need to `cleanup()` in this path too?